src/components/OutSegTemplateEditor.jsx

```diff
@@ -83,6 +87,20 @@ const OutSegTemplateEditor = memo(({ outSegTemplate, setOutSegTemplate, generate
 
   const needToShow = shown || error != null;
 
+  const onVariableClick = useCallback((variable) => {
+    const input = inputRef.current;
+    const startPos = input.selectionStart;
+    const endPos = input.selectionEnd;
+
+    const newValue = `${text.slice(0, startPos)}${`\${${variable}}${text.slice(endPos)}`}`;
+    setText(newValue);
+
+    // Move the cursor to after the inserted variable
+    const newPos = startPos + variable.length + 2;
+    input.selectionStart = newPos;
+    input.selectionEnd = newPos;
```

Comment:
I removed this code because I cannot get it to work. Everytime i click one of the variable buttons, the input loses focus. if I give focus again, the cursor position is gone.
let me know if Im missing something.

Reply:
You're right, there is some code here in my original PR that didn't 100% work that I didn't notice before pushing.

Overall, the code seems to work like how I expect after I pulled your changes; the user can still inject the button based on cursor position, even if the input focus is lost afterwards.

I checked your changes on my end here: https://i.imgur.com/bnX8Ds4.mp4

So, from my perspective everything is still working as I expect, and thanks for removing the extraneous code that doesn't do anything (to be honest, I used ChatGPT a little to help me out as I'm not that familiar with ReactJS, so there may be some extra boilerplate sorry).